File: back/Models/Message.cs
// Models/Message.cs
namespace MatchJob.Models;

public class Message
{
    public long Id { get; set; }

    public long ConversationId { get; set; }
    public Conversation Conversation { get; set; } = null!;

    public long SenderId { get; set; }
    public User Sender { get; set; } = null!;

    public string Content { get; set; } = string.Empty;

    // Preenchido automaticamente pelo EF ao inserir
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: mobile/src/types/message.ts
export interface Conversation {
  Id: string;
  ClientId: string;
  ClientName: string;
  ProfessionalId: string;
  ProfessionalName: string;
}

export interface Message {
  Id: string;
  ConversationId: string;
  SenderId: string;
  SenderName: string;
  Content: string;
  CreatedAt: string;
}

export interface CreateConversationRequest {
  ClientId: string;
  ProfessionalId: string;
}

export interface SendMessageRequest {
  ConversationId: string;
  SenderId: string;
  Content: string;
}
